src/transformers/typescript.ts

```diff
@@ -144,7 +144,10 @@ const transformer: Transformer<Options.Typescript> = ({
     sourceMapText: map,
     diagnostics,
   } = ts.transpileModule(content, {
-    fileName: filename,
+    fileName:
+      filename.slice(0, 1) === '.'
```

Comment:
replace with `!isAbsolute(filename)` (from `path`)

Reply:
Applied in https://github.com/sveltejs/svelte-preprocess/commit/842867034f90c12d6a5fc523477133c8117b5d38. Thank you!